### ISSUE TITLE
Rescan Single Album rather than whole Artist

### ISSUE THREAD

**Issue by blade316** [lidarr/Lidarr#492]
**Is your feature request related to a problem? Please describe.**
At the moment it seems you can only rescan files for an entire artist, rather than just rescanning files for a specific album. An example I have is the album 'Cinema' by 'Andrea Bocelli', I have the entire album, however it only thinks I have half of it, so I would like to force a refresh, but only for that album.

![image](https://user-images.githubusercontent.com/19606551/45596447-5b0a4980-b9ff-11e8-928a-e88216d6f157.png)

![image](https://user-images.githubusercontent.com/19606551/45596469-8725ca80-b9ff-11e8-9797-9eb767da7b44.png)

**Describe the solution you'd like**
Would be good to have an additional option is the album view, for something like 'Scan/Refresh Album Files'. This then rescans the files and adds missing ones etc.
 
**Describe alternatives you've considered**
Using the 'Refresh and Scan' option, but this is for the whole artist, and takes a very long time if you simply want to refresh a single album.

**Additional context**
None.

**Comment by Qstick** [comment]
We don’t associate an album to a directory location thus the only reference we have to scan is the artist folder. This allows us to stay flexible with whatever way you want to organize and name your songs.

**Comment by blade316** [comment]
@Qstick hmmm ok, I can sort of see the mindset behind that decision, however do you see a benefit in allowing to scan single individual album folders within an artist folder, rather than having to re-scan the entire artist.

Take any artist with a lot of albums and your scan time for a simple track or single album change, goes from a 20-30 seconds to 10-15mins.

If you can suggest a better way to scan individual album folders for changes, I’m more than happy to try that?

**Comment by Qstick** [comment]
10-15 min seems excessive for a Artist folder scan. For artists that take that long, how many tracks are there? How many other metadata files? Can you upload a log file of a single artist scan that is taking that long? Im not sure I’ve seen one go longer than 1-2 minutes, maybe I don’t have enough music though :)

**Comment by blade316** [comment]
Yep no worries I can do that, I’m just off to work now so I’ll do it when I get home this evening 👍🏼

**Comment by Qstick** [comment]
Sounds good, just as a reference I just ran a test import of 111 artists and it only took 12 minutes to grab all the data from online, scan all artist folders for tracks and metadata, and match up tracks.

**Comment by blade316** [comment]
Do you have any artists in your library such as Michael Jackson, Elvis Presley, the Beatles, Frank Sinatra, Dean Martin, Elton John, Phil Collins, Elvis Costello, Eric Clapton, AC/DC, Metallica, Red Hot Chili Peppers (off the top of my head) .... these will have a lot of albums, I’d be curious to know if the timeline is the same

**Comment by Qstick** [comment]
Yes, three questions on your settings that I think could be slowing things down:

What do your metadata profiles look like?
Do you have analize audio files turned on?
How much metadata do you have? (nfo, jpg, lyric files, etc)

**Comment by blade316** [comment]
My metadata profile is:

Album

Studio
Live
Compilation

I have changed this and noticed neglible speed increases

I didn’t even know analyse audio files was an option, where is it in the settings? And I look when I get home.

I’ve tried with just existing NFO and folder/cover jpg and without and existing data, and  noticed no speed increases.

SIDE NOTE: most of my files are FLAC files, and at the moment, the majority of them get parsing failures due to either “cant find album (or similar)” or “Unable to find file” which isn’t right because the file is there .... or it seems to recognise FLAC files as MP3-320 files.

separate issue but I thought I would mention it. I have seen a few posts about FLAC issues, but I got the impression it was fixed in a previous release.

I’m running nightly btw.
